design-documents/Remove-__sleep-and-__wakeup.md

```diff
@@ -0,0 +1,35 @@
+### Remove __sleep and __wakeup methods from our code
+#### Why?
+* Serializing objects requires classes to always consider serialization process
+and it is easy for multiple developers working on the same class to forget about it.
+* PHP serialization opens up our code to potential security issues (Remote code execution).
+* During serialization objects of classes that do not expect serialization may be serialized
+anyway causing unexpected behaviour.
+
+#### Current situation
+The only place we serialize objects (as far as I am aware of) is storing User and Acl objects
```

Comment:
Any object stored in the session should be serializable

Reply:
We should avoid storing objects in session, do you agree?